src/rgw/rgw_iam_managed_policy.h

```diff
@@ -36,4 +43,112 @@ struct ManagedPolicies {
 void encode(const ManagedPolicies&, bufferlist&, uint64_t f=0);
 void decode(ManagedPolicies&, bufferlist::const_iterator&);
 
+struct ManagedPolicyAttachment {
+  std::string arn;
+  std::string  status;
```

Comment:
in the design doc, this attachment status was represented as an enum. that would be more space-efficient than storing as a string